sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/implementation/AccessTokenCache.java

```diff
@@ -0,0 +1,299 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.experimental.implementation;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.experimental.credential.PopTokenRequestContext;
+import com.azure.core.experimental.credential.SupportsProofOfPossession;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.logging.LogLevel;
+import com.azure.core.util.logging.LoggingEventBuilder;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.Signal;
+import reactor.core.publisher.Sinks;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+/**
+ * A token cache that supports caching a token and refreshing it.
+ */
+public final class AccessTokenCache {
```

Comment:
Will this get folded into the existing `AccessTokenCache`? or should we name this `PopAccessTokenCache` or something?

Reply:
It will be folded